lang/ruby/lib/avro/io.rb

```diff
@@ -31,6 +31,49 @@ def initialize(writers_schema, readers_schema)
       end
     end
 
+    def self.match_schemas(writers_schema, readers_schema)
+      w_type = writers_schema.type_sym
+      r_type = readers_schema.type_sym
+
+      # This conditional is begging for some OO love.
+      if w_type == :union || r_type == :union
+        return true
+      end
+
+      if w_type == r_type
+        return true if Schema::PRIMITIVE_TYPES_SYM.include?(r_type)
+
+        case r_type
+        when :record
+          return writers_schema.fullname == readers_schema.fullname
```

Comment:
Aren't the `return` statements in this `case` unnecessary, since the `case` will evaluate to the desired expression?